src/solvers/flattening/flatten_byte_operators.cpp

```diff
@@ -199,8 +202,6 @@ exprt flatten_byte_extract(
   // big-endian: (short)concatenation(01,02)=0x0102
   // little-endian: (short)concatenation(03,04)=0x0304
 
-  assert(src.operands().size()==2);
```

Comment:
which check covers this?

Reply:
The fact that the parameter type is `byte_extract_exprt`.